src/main/python/systemml/mllearn/estimators.py

```diff
@@ -392,13 +392,17 @@ def save(self, outputDir, format='binary', sep='/'):
         """
         if self.model != None:
             self.model.save(self.sc._jsc, outputDir, format, sep)
-            if self.le is not None:
+
+            labelMapping = None
+            if hasattr(self, 'le') and self.le is not None:
                 labelMapping = dict(enumerate(list(self.le.classes_), 1))
-            else:
+            elif hasattr(self, 'labelMap') and self.labelMap is not None:
                 labelMapping = self.labelMap
-            lStr = [ [ int(k), str(labelMapping[k]) ] for k in labelMapping ]
-            df = self.sparkSession.createDataFrame(lStr)
-            df.write.csv(outputDir + sep + 'labels.txt', mode='overwrite', header=False)
+
+            if labelMapping is not None:
+                lStr = [ [ int(k), str(labelMapping[k]) ] for k in labelMapping ]
+                df = self.sparkSession.createDataFrame(lStr)
+                df.write.csv(outputDir + sep + 'labels.txt', mode='overwrite', header=False)
```

Comment:
OK, I can add it later, probably u added comment at the same time I was pushing the change.